### DISCUSSION TITLE
How to create application using Clean Architecture and EF DB First Approach

### DISCUSSION THREAD

**Discussion by milyas-salik** [jasontaylordev/CleanArchitecture#559] [General]
Hi,

I have seen plenty of examples of clean architecture with EF Code first. But I don't prefer code first and not convenient to go with it instead I have my database with tables . I would like to design architecture with EF DB first approach using clean architecture.

Please let me know the suitable way or steps to proceed !

**Comment by yaseerngr** [comment]
I do follow this template and use DB first approach. 

- In the infrastructure/persistence layer, add your DB context as running the scaffold-DbContext command. This will create a massive ...dbcontext.cs file and Db models. 
- Move all the DB models to Domain/Entities
- Extract configurations from the larger ..dbcontext.cs file and put it under Infrastructure/Persistance/Configurations . You can write a small program to extract this configurations. I used this this project to extract https://github.com/lauxjpn/DbContextOnModelCreatingSplitter

**Comment by hotfirewire** [comment]
I've done something similar and I've used the [EF Core Power Tools](https://marketplace.visualstudio.com/items?itemName=ErikEJ.EFCorePowerTools).  The tool has the ability to separate the configurations into separate files and even place your models in the Domain project for you.  It saves everything to a config file for easy repeatability.
